### PR TITLE
Bump Quay to 3.6.4

### LINKED ISSUE #44: Absence of `quayHostname` or `targetHostname` should default to local hostname
When running `mirror-registry` without either the `--quayHostname` or `--targetHostname` parameter, it currently defaults to `localhost`. This is not useful for a registry that is supposed to be used by external clients for pulling images. 

Instead the default should be to fallback to the output of `hostname -f`.

### PR DESCRIPTION
**Pull request by HammerMeetNail**
* Testing upgrades from 1.0 to 1.1 with a Quay version bump
* Updates default hostname, fixes #44

```
[doconnor@doconnor-omr-dev ~]$ ./mirror-registry upgrade -v

   __   __
  /  \ /  \     ______   _    _     __   __   __
 / /\ / /\ \   /  __  \ | |  | |   /  \  \ \ / /
/ /  / /  \ \  | |  | | | |  | |  / /\ \  \   /
\ \  \ \  / /  | |__| | | |__| | / ____ \  | |
 \ \/ \ \/ /   \_  ___/  \____/ /_/    \_\ |_|
  \__/ \__/      \ \__
                  \___\ by Red Hat
 Build, Store, and Distribute your Containers

INFO[2022-03-04 17:54:12] Install has begun
DEBU[2022-03-04 17:54:12] Ansible Execution Environment Image: quay.io/quay/mirror-registry-ee:latest
DEBU[2022-03-04 17:54:12] Pause Image: registry.access.redhat.com/ubi8/pause:latest
DEBU[2022-03-04 17:54:12] Quay Image: registry.redhat.io/quay/quay-rhel8:v3.6.2
DEBU[2022-03-04 17:54:12] Redis Image: registry.redhat.io/rhel8/redis-6:1-25
DEBU[2022-03-04 17:54:12] Postgres Image: registry.redhat.io/rhel8/postgresql-10:1-161
...
INFO[2022-03-04 17:55:38] Quay upgraded successfully
```

### CODE REVIEW

File: ansible-runner/context/app/project/roles/mirror_appliance/tasks/create-init-user.yaml [hunk 1]
```diff
@@ -7,3 +7,4 @@
     body_format: json
     body: '{ "username": "{{ init_user }}", "password": "{{ init_password }}", "email": "init@quay.io", "access_token": "true" }'
   register: result
+  ignore_errors: true # temp hack to test out upgrades
```
Comment (jonathankingfc) [inline ansible-runner/context/app/project/roles/mirror_appliance/tasks/create-init-user.yaml:10]: Is this going to be in the final PR?

File: cmd/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+func init() {
+
+	// Add upgrade command
+	rootCmd.AddCommand(upgradeCmd)
+
+	upgradeCmd.Flags().StringVarP(&targetHostname, "targetHostname", "H", getFQDN(), "The hostname of the target you wish to install Quay to. This defaults to $HOST")
+	upgradeCmd.Flags().StringVarP(&targetUsername, "targetUsername", "u", os.Getenv("USER"), "The user on the target host which will be used for SSH. This defaults to $USER")
+	upgradeCmd.Flags().StringVarP(&sshKey, "ssh-key", "k", os.Getenv("HOME")+"/.ssh/quay_installer", "The path of your ssh identity key. This defaults to ~/.ssh/quay_installer")
+
+	upgradeCmd.Flags().StringVarP(&quayHostname, "quayHostname", "", "", "The value to set SERVER_HOSTNAME in the Quay config.yaml. This defaults to <targetHostname>:8443")
```
Comment (jonathankingfc) [inline cmd/upgrade.go:34]: Do we need this in the upgade?

File: Dockerfile [hunk 3]
```diff
@@ -54,7 +54,7 @@ RUN /output/install-from-bindep && rm -rf /output/wheels
 COPY ansible-runner/context/app /runner
 
 # Pull in Quay dependencies
-FROM registry.redhat.io/quay/quay-rhel8:v3.6.1 as quay
+FROM registry.redhat.io/quay/quay-rhel8:v3.6.2 as quay
```
Comment (jonathankingfc) [inline Dockerfile:57]: Is there a way for us to have a single source of truth for the image versions?

File: ansible-runner/context/app/project/roles/mirror_appliance/tasks/create-init-user.yaml (hunk 1 above)
Comment (HammerMeetNail) [reply ansible-runner/context/app/project/roles/mirror_appliance/tasks/create-init-user.yaml:10]: Nope, I'll remove that.

File: cmd/upgrade.go (hunk 2 above)
Comment (HammerMeetNail) [reply cmd/upgrade.go:34]: I think so...if we remove it, there wouldn't be any way for users who remotely installed mirror-registry to upgrade their remote installations. Need the SSH info.

File: Dockerfile (hunk 3 above)
Comment (HammerMeetNail) [reply Dockerfile:69]: Fixed. Everything is set in `.env`

**Review by jonathankingfc — APPROVED** [review]
LGTM

**Pull request merged**